src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -258,6 +259,22 @@ class ArrayBuffer[A] private (initialElements: Array[AnyRef], initialSize: Int)
     }
     this
   }
+
+  @tailrec private def foldl[B](start: Int, end: Int, z: B, op: (B, A) => B): B =
+    if (start == end) z
+    else foldl(start + 1, end, op(z, array(start).asInstanceOf[A]), op)
```

Comment:
Is this more efficient than the version for IndexedSeqOps because it avoids the indirection through `apply`? Does that difference show up in benchmarks?

Reply:
Yes, I saw a diff in benchmark, but I did not analyze. I mention it somewhere.